File: app/schemas/location.py
from pydantic import BaseModel, ConfigDict
from datetime import datetime


class LocationShareRequestSchema(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    phone: str
    created_at: datetime
    expired_at: datetime


class LocationShareRecordSchema(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    request_id: int
    latitude: float
    longitude: float
    created_at: datetime

File: frontend/src/types/location.ts
export interface LocationShareRequest {
  id: number;
  phone: string;
  created_at: string;
  expired_at: string;
}

export interface LocationShareRecord {
  id: number;
  request_id: number;
  latitude: number;
  longitude: number;
  created_at: string;
}

export interface LocationShareRecordList {
  items: LocationShareRecord[];
}